node/builder.go

```diff
@@ -586,14 +586,45 @@ func Repo(r repo.Repo) Option {
 			return err
 		}
 
+		var cfg *config.Blockstore
+		switch settings.nodeType {
+		case repo.FullNode:
+			cfgp, ok := c.(*config.FullNode)
+			if !ok {
+				return xerrors.Errorf("invalid config from repo, got: %T", c)
+			}
+			cfg = &cfgp.Blockstore
+		case repo.StorageMiner:
+			cfgp, ok := c.(*config.StorageMiner)
+			if !ok {
+				return xerrors.Errorf("invalid config from repo, got: %T", c)
+			}
+			cfg = &cfgp.Blockstore
+		default:
+			cfg = &config.Blockstore{}
+		}
+
 		return Options(
 			Override(new(repo.LockedRepo), modules.LockedRepo(lr)), // module handles closing
 
 			Override(new(dtypes.MetadataDS), modules.Datastore),
 			Override(new(dtypes.UniversalBlockstore), modules.UniversalBlockstore),
-			Override(new(dtypes.ChainBlockstore), modules.ChainBlockstore),
-			Override(new(dtypes.StateBlockstore), modules.StateBlockstore),
-			Override(new(dtypes.ExposedBlockstore), From(new(dtypes.UniversalBlockstore))),
+
+			If(cfg.EnableSplitstore,
+				If(cfg.Splitstore.GetHotStoreType() == "badger",
+					Override(new(dtypes.HotBlockstore), modules.BadgerHotBlockstore)),
+				Override(new(dtypes.SplitBlockstore), modules.SplitBlockstore(cfg)),
+				Override(new(dtypes.ChainBlockstore), modules.ChainSplitBlockstore),
+				Override(new(dtypes.StateBlockstore), modules.StateSplitBlockstore),
+				Override(new(dtypes.BaseBlockstore), From(new(dtypes.SplitBlockstore))),
+				Override(new(dtypes.ExposedBlockstore), From(new(dtypes.SplitBlockstore))),
+			),
+			If(!cfg.EnableSplitstore,
+				Override(new(dtypes.ChainBlockstore), modules.ChainFlatBlockstore),
+				Override(new(dtypes.StateBlockstore), modules.StateFlatBlockstore),
+				Override(new(dtypes.BaseBlockstore), From(new(dtypes.UniversalBlockstore))),
+				Override(new(dtypes.ExposedBlockstore), From(new(dtypes.UniversalBlockstore))),
+			),
```

Comment:
we'd still need it because we need to pass the splitstore config to the constructor.

Reply:
Or do you want to not have a blockstore config section at all and put the Splitstore config in common?

Reply:
I actually would like to keep the blockstore config section since we are planning to build nodes that operate without a coldstore in the future.

But if you feel strongly about it, we can certainly put it in ConfigCommon.